createHandler.js

```diff
@@ -0,0 +1,328 @@
+        -1
+    )
+    .filter(handlerTag => handlerTag > 0);
+}
+
+function hasUnresolvedRefs(props) {
+  const extract = refs => {
+    if (!Array.isArray(refs)) {
+      return refs && refs.current === null;
+    }
+    return refs.some(r => r && r.current === null);
+  };
+  return extract(props['simultaneousHandlers']) || extract(props['waitFor']);
+}
+
+const stateToPropMappings = {
+  [State.BEGAN]: 'onBegan',
+  [State.FAILED]: 'onFailed',
+  [State.CANCELLED]: 'onCancelled',
+  [State.ACTIVE]: 'onActivated',
+  [State.END]: 'onEnded',
+};
+
+export default function createHandler(
+  handlerName,
+  propTypes = {},
+  config = {},
+  transformProps,
+  customNativeProps = {}
+) {
+  class Handler extends React.Component {
+    static displayName = handlerName;
+
+    static propTypes = propTypes;
+
+    constructor(props) {
+      super(props);
+      this._handlerTag = handlerTag++;
+      this._config = {};
+      if (props.id) {
+        if (handlerIDToTag[props.id] !== undefined) {
+          throw new Error(`Handler with ID "${props.id}" already registered`);
+        }
+        handlerIDToTag[props.id] = this._handlerTag;
+      }
+    }
+
+    _onGestureHandlerEvent = event => {
+      if (event.nativeEvent.handlerTag === this._handlerTag) {
+        this.props.onGestureEvent && this.props.onGestureEvent(event);
+      } else {
+        this.props.onGestureHandlerEvent &&
+          this.props.onGestureHandlerEvent(event);
+      }
+    };
+
+    _onGestureHandlerStateChange = event => {
+      if (event.nativeEvent.handlerTag === this._handlerTag) {
+        this.props.onHandlerStateChange &&
+          this.props.onHandlerStateChange(event);
+
+        const stateEventName = stateToPropMappings[event.nativeEvent.state];
+        if (typeof this.props[stateEventName] === 'function') {
+          this.props[stateEventName](event);
+        }
+      } else {
+        this.props.onGestureHandlerStateChange &&
+          this.props.onGestureHandlerStateChange(event);
+      }
+    };
+
+    _refHandler = node => {
+      this._viewNode = node;
+
+      const child = React.Children.only(this.props.children);
+      const { ref } = child;
+      if (ref !== null) {
+        if (typeof ref === 'function') {
+          ref(node);
+        } else {
+          ref.current = node;
+        }
+      }
+    };
+
+    _createGestureHandler = newConfig => {
+      this._config = newConfig;
+
+      RNGestureHandlerModule.createGestureHandler(
+        handlerName,
+        this._handlerTag,
+        newConfig
+      );
+    };
+
+    _attachGestureHandler = newViewTag => {
+      this._viewTag = newViewTag;
+
+      RNGestureHandlerModule.attachGestureHandler(this._handlerTag, newViewTag);
+    };
+
+    _updateGestureHandler = newConfig => {
+      this._config = newConfig;
+
+      RNGestureHandlerModule.updateGestureHandler(this._handlerTag, newConfig);
+    };
+
+    _dropGestureHandler = () => {
```

Comment:
Why is it not used?

Reply:
What do you mean?

Reply:
`_dropGestureHandler` is not used anywhere. Isn't it a bug?

Reply:
Yep, good catch! Sorry..

Reply:
np, thanks!